src/app/components/auth/login/login.component.ts

```diff
@@ -79,4 +80,14 @@ export class LoginComponent {
   goToHome(): void {
     this.router.navigate([RoutesConstants.home]);
   }
+  public togglePasswordVisibility(): void {
+    this.showPassword = !this.showPassword;
+  }
+  isFieldInvalid(field: string): boolean {
+    const control = this.loginForm.get(field);
+    return control?.invalid && (control.dirty || control.touched)
+      ? true
+      : false;
```

Comment:
Is not necessary the ternary operator 
```suggestion
    return control?.invalid && (control.dirty || control.touched);
```